diff --git a/include/cst_file.h b/include/cst_file.h
index eaf2c4a..b141cfd 100644
--- a/include/cst_file.h
+++ b/include/cst_file.h
@@ -99,6 +99,9 @@ long cst_fread(cst_file fh, void *buf, long size, long count);
 int cst_fprintf(cst_file fh, const char *fmt, ...);
 int cst_sprintf(char *s, const char *fmt, ...);
 #ifdef _WIN32
+#include <windows.h>
+/* Not needed if this is mingw-w64 */
+#ifndef __MINGW64_VERSION_MAJOR
 #define snprintf c99_snprintf
 
 #ifdef __GNUC__
@@ -127,6 +130,7 @@ INLINE int c99_snprintf(char* str, size_t size, const char* format, ...)
        return count;
    }
 #endif
+#endif
 #define cst_snprintf snprintf
 
 #if defined(__palmos__)
diff --git a/main/Makefile b/main/Makefile
index 8166182..a4fb491 100644
--- a/main/Makefile
+++ b/main/Makefile
@@ -152,12 +152,12 @@ $(BINDIR)/flite_${VOICE}: flite_main.o flite_lang_list $(flite_LIBS_deps)
 
 install:
 #       The basic binaries
-	@ $(INSTALL) -m 755 $(BINDIR)/flite $(DESTDIR)$(INSTALLBINDIR)
+	@ $(INSTALL) -m 755 $(BINDIR)/flite$(EXEEXT) $(DESTDIR)$(INSTALLBINDIR)
 	@ for i in $(VOICES) ; \
 	do \
-	   $(INSTALL) $(BINDIR)/flite_$$i $(DESTDIR)$(INSTALLBINDIR); \
+	   $(INSTALL) $(BINDIR)/flite_$$i$(EXEEXT) $(DESTDIR)$(INSTALLBINDIR); \
 	done
-	$(INSTALL) -m 755 $(BINDIR)/flite_time $(DESTDIR)$(INSTALLBINDIR)
+	$(INSTALL) -m 755 $(BINDIR)/flite_time$(EXEEXT) $(DESTDIR)$(INSTALLBINDIR)
 #       The libraries: static and shared (if built)
 	cp $(CP_FLAGS) $(flite_LIBS_deps) $(DESTDIR)$(INSTALLLIBDIR)
 ifdef SHFLAGS
diff --git a/src/utils/cst_tokenstream.c b/src/utils/cst_tokenstream.c
index a903044..9b2a4c0 100644
--- a/src/utils/cst_tokenstream.c
+++ b/src/utils/cst_tokenstream.c
@@ -267,7 +267,7 @@ static void get_token_sub_part(cst_tokenstream *ts,
 }
 
  #if defined _WIN32 && !defined(__GNUC__)
- __inline int ts_utf8_sequence_length(char c0)
+ int ts_utf8_sequence_length(char c0)
  #else
  int ts_utf8_sequence_length(char c0)
  #endif
diff --git a/src/utils/cst_val.c b/src/utils/cst_val.c
index 40cf1e9..5603ce2 100644
--- a/src/utils/cst_val.c
+++ b/src/utils/cst_val.c
@@ -476,7 +476,7 @@ int val_dec_refcount(const cst_val *b)
 }
 
 #ifdef _WIN32
-__inline int utf8_sequence_length(char c0)
+int utf8_sequence_length(char c0)
 #else
 int utf8_sequence_length(char c0)
 #endif
